### ISSUE TITLE
[Feature Request] More Defense Technology

### ISSUE THREAD

**Issue by SupernovaCo** [aidancbrady/DefenseTech#15]
I've just been fooling around with this mod (and having a lot of fun reliving my Voltz days), and I had a few ideas.
Now, I know that everyone's favorite part of ICBM was (mostly) the missiles, and that's what DefenseTech aims to do, but I think it would be really  cool if DefenseTech went for more of a general  Voltz-esque defense technology style, with features like turrets (OpenModularTurrets 2 is good, but I still like the old ICBM turrets better, with their realistic lasers, animated rotating turrets and whatnot), forcefields, security cameras (because why not; possibly use XCompWiz's LookingGlass API), fences and blastproof blocks, sensors (for detecting players and enemies and such), and so on.

I'd gladly help provide ideas and assets (I.E. textures, models, sounds, etc. I'm fairly good with modeling and texturing, and already work as a modeler for a mod), and so on. If this doesn't fit with your goal for the mod, that's fine, I just though it would be neat to have this stuff, especially since I'm working on creating a mod pack called Megawatt that serves as a sort of spiritual successor to Voltz, which will be using DefenseTech.

**Comment by Robot-Gentleman** [comment]
There is already a force feild mod for 1.7 from calclavia and OpenModularTurrets has blast proof blocks fyi.

**Comment by Robot-Gentleman** [comment]
But I do think it would be great if the sensors and railguns were implemented again.

**Comment by SupernovaCo** [comment]
Yep. I knew that when I wrote this post.
Also, I liked the Calclavia forcefields, but I think it could be done
better. ;)

*--Marvin_Megavolt*

**Comment by SupernovaCo** [comment]
Also, another idea: An alarm siren with lots of different sounds to pick from, and animated alarm lights that project visible 'beams' of light when on (actually just 2D textures resembling beams of light). Just for base aesthetics.

**Comment by buithanh12072001** [comment]
Yep, your request is a good idea. And other request: Some complex machine to make Antimatter, which is used for Antimatter or Red Matter missile

**Comment by SupernovaCo** [comment]
He's already got plans for adding some sort of endgame machines to Mekansim for the antimatter and red matter i think.

**Comment by SupernovaCo** [comment]
A couple of other ideas: Firstly, gas vents: Emit gases (kinda like the various gas explosives from DefenseTech's classic ICBM part) which are Mekanism gases; used to defend your base by flooding sections of y our base with gas. Second, redesign the recipes for the explosives to be generally somewhat better (I will post some ideas).

**Comment by SupernovaCo** [comment]
Here's a sound I made for the force fields: 
[Forcefield_01.zip](https://github.com/aidancbrady/DefenseTech/files/245710/Forcefield_01.zip)

I imagine this sound being audible to players who are near the shield projector and the shield bubble itself.

**Comment by Robot-Gentleman** [comment]
So kind of like Glen's Gasses except not as buggy and more futuristic-ish. If that was implemented I would like the emp to disable the seals on gas containers, filling peoples bases with gas

**Comment by SupernovaCo** [comment]
Also, here's a texture for the shield blocks that compose the forcefield:
[forcefield.zip](https://github.com/aidancbrady/DefenseTech/files/245798/forcefield.zip)
It's white because I imagine the forcefields having a customizable color.

**Comment by SupernovaCo** [comment]
And to you Robot-Gentleman, yes! I really like that idea with the EMP setting off all the gas traps. >:D

**Comment by buithanh12072001** [comment]
It's a god idea. Another request: I want the end game machine that create anti/red matter must be complex. It can be a multi-block structure, or even the same as real-life LHC

**Comment by SupernovaCo** [comment]
Like the old Atomic Science particle acceler, with its quirky and often hilarious unreliability. :)

**Comment by SupernovaCo** [comment]
Suggestion for the Forcefield Projectors: Make them some sort of big multiblock structure ( with a custom 3D model perhaps?)

**Comment by buithanh12072001** [comment]
Great idea

**Comment by LemADEC** [comment]
or you could use the force field system from WarpDrive :)

**Comment by CaptainMegavolt** [comment]
There is one?

_--Marvin_Megavolt_

**Comment by CaptainMegavolt** [comment]
And, no offense, but I quite liked MFFS.

_--Marvin_Megavolt_

**Comment by LemADEC** [comment]
Then don't try the alternative if you start by saying it's crap without looking into it...

**Comment by CaptainMegavolt** [comment]
I do not plan on saying anything is crap (I'm not that sort of guy to
unrightfully insult something), and I believe that, if it was made by the
WarpDrive team, it can't possibly be bad...

_--Marvin_Megavolt_
